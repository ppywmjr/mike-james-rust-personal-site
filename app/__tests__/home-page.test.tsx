import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";

import Home from "../page";

describe("Home page", () => {
  it("renders with h1", () => {
    render(<Home />);

    const heading = screen.getByRole("heading", { level: 1 });

    expect(heading).toBeVisible();
  });

  it("renders with img", () => {
    render(<Home />);

    const image = screen.getByRole("img", { name: /My selfie/ });

    expect(image).toBeVisible();
  });

  it("renders with link card", () => {
    render(<Home />);

    const text = screen.getByRole("link", { name: /Me and my site/ });

    expect(text).toBeVisible();
  });
});
