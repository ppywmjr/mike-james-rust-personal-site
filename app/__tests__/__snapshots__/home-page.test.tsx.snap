// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders the page 1`] = `
<main
  className="flex flex-col justify-between items-center p-12 pt-1 min-h-screen"
>
  <div
    className="flex flex-row self-stretch items-center justify-between"
  >
    <h1
      className="font-bold"
    >
      Mike James Rust - a QA's playground
    </h1>
    <div>
      <div
        className="px-4 flex flex-row items-center"
      >
        <img
          alt="Mike James Rust Logo"
          data-nimg="1"
          decoding="async"
          fetchpriority="high"
          height={6}
          onError={[Function]}
          onLoad={[Function]}
          src="/logo.svg"
          style={
            {
              "color": "transparent",
            }
          }
          width={25}
        />
      </div>
    </div>
  </div>
  <div>
    <img
      alt="My selfie. A handsome chap with long wavy hair, a tidy beard and a big smile"
      className="rounded-2xl"
      data-nimg="1"
      decoding="async"
      fetchpriority="high"
      height={214}
      onError={[Function]}
      onLoad={[Function]}
      src="/_next/image?url=%2Fprofile.jpg&w=384&q=75"
      srcSet="/_next/image?url=%2Fprofile.jpg&w=256&q=75 1x, /_next/image?url=%2Fprofile.jpg&w=384&q=75 2x"
      style={
        {
          "color": "transparent",
        }
      }
      width={180}
    />
  </div>
  <div
    className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3"
  >
    <a
      className="p-4 max-w-[16rem] group"
      href="/me-and-my-site"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      <h2
        className="font-bold my-4"
      >
        Me and my site
         
        <span
          className="inline-block transition-all group-hover:translate-x-1"
        >
          -&gt;
        </span>
      </h2>
      <p
        className="text-sm"
      >
        A first page, with a little about who I am and why I'm doing this.
      </p>
    </a>
    <a
      className="p-4 max-w-[16rem] group"
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      <h2
        className="font-bold my-4"
      >
        Quality topics
         
        <span
          className="inline-block transition-all group-hover:translate-x-1"
        >
          -&gt;
        </span>
      </h2>
      <p
        className="text-sm"
      >
        THIS DOES NOT EXIST YET
      </p>
    </a>
    <a
      className="p-4 max-w-[16rem] group"
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      <h2
        className="font-bold my-4"
      >
        The making of
         
        <span
          className="inline-block transition-all group-hover:translate-x-1"
        >
          -&gt;
        </span>
      </h2>
      <p
        className="text-sm"
      >
        THIS DOES NOT EXIST YET
      </p>
    </a>
  </div>
</main>
`;
